R/publish.R

```diff
@@ -8,23 +8,50 @@
 #' @param account Account name to publish to. Will default to any previously published
 #'  to account or any single account already associated with \code{server}.
 #' @param server Server to publish to (by default beta.rstudioconnect.com
-#' but any RStudio Connect server can be published to).
+#'  but any RStudio Connect server can be published to).
+#' @param render \code{TRUE} to render all formats prior to publishing (defaults to
+#'  \code{FALSE}, however this can be modified via the \code{bookdown.render_on_publish}
+#'  option). Note that this requires the use of a Makefile to provide the
+#'  implementaiton of rendering all formats.
 #'
 #' @export
 publish_book <- function(
-  name = NULL, account = NULL, server = 'beta.rstudioconnect.com'
+  name = NULL, account = NULL, server = NULL,
+  render = getOption("bookdown.render_on_publish", FALSE)
 ) {
 
   on.exit(opts$restore(), add = TRUE)
+
+  # render if requested (not we use the Makefile here rather than calling
+  # render_book because we need some mechanism that will have a-priori
+  # knowledge of all book formats)
+  if (isTRUE(render)) {
+    if (file.exists("Makefile")) {
+      result <- system2("make")
+      if (result != 0)
+        stop("Error ", result, " attempting to render book")
+    }
+    else
+      stop("Rendering before publishing requires a Makefile")
```

Comment:
Perhaps just call `render_book('index.Rmd')` here? Requiring Makefile may be an issue for Windows users.